psd-web/app/models/user.rb

```diff
@@ -151,4 +152,8 @@ def has_viewed_introduction!
   def current_user?
     User.current&.id == id
   end
+
+  def has_role?(role)
+    user_roles.where(name: role).count.positive?
```

Comment:
[exists?](https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-exists-3F) might be marginally faster: 

```suggestion
    user_roles.exists?(name: role)
```